### ISSUE TITLE
Fix S2068 FP: Ignore common safe words

### ISSUE THREAD

**Issue by hendrik-buchwald-sonarsource** [SonarSource/sonar-dotnet#7914]
### Description

Rule [S2068](https://sonarsource.github.io/rspec/#/rspec/S2068) searches for variable names of hard-coded strings that contain certain keywords like `password`. This can easily result in false positives since there are valid use cases to have a hard-coded string that relates to passwords but is not a password.

### Repro steps

```
const string passwordQuestion = "What is your favorite color";
```

```
<credentials passwordFormat="Clear"></credentials>
```

### Expected behavior

No issue should be raised as the variable does not contain a password.

### Actual behavior

An issue is raised.

### Known workarounds

The detection could be improved by adding an exclude-list additionally to the current include-list. This exclude-list should be configurable by the user, the same as the include-list. By default, it could contain the following values:

 * passwordQuestion
 * passwordFormat

### Related information

* https://sonarsource.atlassian.net/browse/APPSEC-1029 (WebGoat.Net)

**Comment by pierre-loup-tristant-sonarsource** [comment]
The FP in `<credentials passwordFormat="Clear"></credentials>` is detected by SonarXML.

**Comment by hendrik-buchwald-sonarsource** [comment]
After the review of this issue, I decided to withdraw this request for change. It is currently not worth the effort for the few false positives that it would solve.
